components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+        final EventContext anotherUser = newUserAndGroup(groupPermissions);
+        /* create a Dataset as the normalUser and import into it */
+        loginUser(normalUser);
+        Dataset dat = mmFactory.simpleDataset();
+        Dataset sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+        final RString imageName = omero.rtypes.rstring(fakeImageFile.getName());
+        final List<List<RType>> result = iQuery.projection(
+                "SELECT id FROM OriginalFile WHERE name = :name ORDER BY id DESC LIMIT 1",
+                new ParametersI().add("name", imageName));
+        final long previousId = result.isEmpty() ? -1 : ((RLong) result.get(0).get(0)).getValue();
+        List<String> path = Collections.singletonList(fakeImageFile.getPath());
+        importFileset(path, path.size(), sentDat);
+        final List<RType> resultAfterImport = iQuery.projection(
+                "SELECT id, details.group.id FROM OriginalFile o WHERE o.id > :id AND o.name = :name",
+                new ParametersI().addId(previousId).add("name", imageName)).get(0);
+        final long remoteFileId = ((RLong) resultAfterImport.get(0)).getValue();
+        long remoteFileGroupId = ((RLong) resultAfterImport.get(1)).getValue();
+        assertOwnedBy((new OriginalFileI(remoteFileId, false)), normalUser);
+        Assert.assertEquals(remoteFileGroupId, normalUser.groupId);
+        Image image = (Image) iQuery.findByQuery(
+                "FROM Image WHERE fileset IN "
+                + "(SELECT fileset FROM FilesetEntry WHERE originalFile.id = :id)",
+                new ParametersI().addId(remoteFileId));
+        /* set up the basic permissions for this test */
+        List<String> permissions = new ArrayList<String>();
+        permissions.add(AdminPrivilegeSudo.value);
+        if (permChown) permissions.add(AdminPrivilegeChown.value);
+
+        final EventContext lightAdmin;
+        lightAdmin = loginNewAdmin(true, permissions);
+        try {
+            sudo(new ExperimenterI(normalUser.userId, false));
+            }catch (SecurityViolation sv) {
+        }
+        /* take care of workflows which do not use sudo */
+        if (!isSudoing) {
+            loginUser(lightAdmin);
+        }
+        /* light admin tries to chown the image of the normalUser whilst sudoed,
+         * which should fail whether they have a Chown permissions or not */
+        client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+        if (isSudoing) {
+            doChange(client, factory, Requests.chown().target(image).toUser(anotherUser.userId).build(), false);
+            long imageGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
+                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
+            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertOwnedBy(image, normalUser);
```

Comment:
I think I addressed this in https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R790 and https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R910 by splitting the testChgrp in two tests and having only one if + else per each of those tests. The content of if and else block is very dissimilar on this latter branch